### DISCUSSION TITLE
Is this the same board with capacitive touch?

### DISCUSSION THREAD

**Discussion by chupocro** [witnessmenow/ESP32-Cheap-Yellow-Display#200] [General]
Looks like it's the same but has a capacitive touch:

https://www.aliexpress.com/item/1005004632953455.html

**Comment by skiphansen** [comment]
Looks like it to me. I just ordered one, if the touch screen isn't supported I'll try to add it.  Looks like a nice board but certainly not the cheapest, but lots of Flash in addition to cap touch.

**Comment by skiphansen** [comment]
I should have done more research before ordering...that board only has 4mb of Flash, not the 32mb I originally thought it had. Oh well.

**Reply by flixmyswitch** [reply]
We have all done that. Have you got this board to work. I am not that good at coding. Trying to find a gui to make the code like square line, but I have been failing badly

**Reply by skiphansen** [reply]
I haven't done anything other than power it up and play with the preflashed demo. The demo is nice and responsive and the cap touch works well.

**Reply by flixmyswitch** [reply]
Yes I found the same. Been playing with square line which is very handy

**Comment by chupocro** [comment]
In the meantime that very board arrived - it's `JC2432W328` and works very well. The documentation from the link in the listing to `JC2432W328.zip` file:

http://pan.jczn1688.com/s/6brhsi

is very good, here are the settings for using it with LVGL and TFT_eSPI:

**User_Setup.h**
```
#define ST7789_DRIVER 
#define TFT_RGB_ORDER TFT_BGR
#define TFT_WIDTH  240
#define TFT_HEIGHT 320
#define TFT_INVERSION_OFF
#define ESP32_DMA
#define TFT_MOSI 13
#define TFT_SCLK 14
#define TFT_CS   15
#define TFT_DC   2
#define TFT_RST  -1
#define TFT_BL   27
#define TOUCH_CS 33
```
**lv_conf.h**
The file from the documentation works well.

**Capacitive touch driver:**
The chip is CST820 and there are `CST820.h` and `CST820.c` files in the documentation. I am using PlatformIO so I just created the `CST820` folder inside `lib` and copied the files.

To test LVGL library I copied the code from `Factory_samples_Capacitive_touch.ino`, added `#include <Arduino.h>`, changed `""` to `<>` in the `#include` directives, uncommented the line `lv_demo_widgets()` which is the main LVGL Widgets demo and it works great.

There are a few more similar Guition boards with different displays and/or with ESP32-S3, here is the link to the documentation:

https://guition.com/doc/%E7%A1%AC%E4%BB%B6%E5%9E%8B%E5%8F%B7.html

Translation to English works well.

As I understand, these boards are designed to be controlled by sending the commands via serial interface and there is `GUITION Online LVGL GUI Design Tool` but I didn't try that.

To backup the factory firmware you can use:
`platformio pkg exec -- esptool.py -b 1500000 --port COMx read_flash 0x000000 0x400000 flash_4M.bin`

To restore the factory firmware:
`platformio pkg exec -- esptool.py -b 1500000 --port COMx write_flash --flash_freq 80m 0x000000 flash_4M.bin`

I added this line to `platformio.ini` to speed up the programming and reading the firmware:
`upload_speed = 1500000`

With the default speed reading of flash took over 6 minutes and after changing the baudrate it took only 35 seconds.
